### PR TITLE
add float matrix using buffer_protocol

### PR DESCRIPTION
**Pull request by DongjiGao**

### CODE REVIEW

File: src/pybind/kaldi_pybind.cc [hunk 1]
```diff
@@ -53,6 +58,27 @@ PYBIND11_MODULE(kaldi_pybind, m) {
...
+      .def("NumCols", &Matrix<float>::NumCols, "Return number of columns")
+      .def("Stride", &Matrix<float>::Stride, "Return stride")
+      .def("__repr__",
+           [] (const Matrix<float> &b) -> std::string {
+             std::ostringstream str; b.Write(str, false); return str.str();
+           })
+      .def(py::init<const MatrixIndexT, const MatrixIndexT,
+          MatrixResizeType, MatrixStrideType>(),
+          py::arg("row"), py::arg("col"), py::arg("resize_type") = kSetZero,
+          py::arg("stride_type") = kDefaultStride);
```
Comment (danpovey) [inline src/pybind/kaldi_pybind.cc:81]: Did you test this from Python?

Comment (DongjiGao) [reply src/pybind/kaldi_pybind.cc:81]: Yes.  I did the same test as vector in README.md.

**Comment by danpovey** [comment]
OK. 
I think if there were a Python test script in that directory it would be good.
That can serve to document the interfaces too.
Can you add one, and then work on wrapping the Kaldi reader/writer objects?  
You'll have to do them one by one, the framework doesn't work with template types directly.

**Comment by DongjiGao** [comment]
> OK.
> I think if there were a Python test script in that directory it would be good.
> That can serve to document the interfaces too.
> Can you add one, and then work on wrapping the Kaldi reader/writer objects?
> You'll have to do them one by one, the framework doesn't work with template types directly.

OK. Is there any example testing script that I can refer to?
For the reader/writer object, should I start with Kaldi IO mechanisms?

**Comment by danpovey** [comment]
You could look at /home/dpovey/filtering/tests/test_multistreamer.py
I am talking about wrapping things like BaseFloatMatrix{Reader,Writer}

**Comment by DongjiGao** [comment]
OK.

**Comment by DongjiGao** [comment]
> You could look at /home/dpovey/filtering/tests/test_multistreamer.py 

There is no such dir and I did not find "test_multistreamer.py" using "find".

**Comment by danpovey** [comment]
sorry, moved it since then, look for lilfilter instead of filtering

**Pull request merged**